pkg/virt-controller/watch/migration.go

```diff
@@ -729,11 +737,54 @@ func (c *MigrationController) handleTargetPodHandoff(migration *virtv1.VirtualMa
 	return nil
 }
 
-func (c *MigrationController) handleSignalMigrationAbort(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance) error {
+func (c *MigrationController) failMigration(isAborted bool, migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance, pod *k8sv1.Pod) {
+	migration.Status.Phase = virtv1.MigrationFailed
+
+	var failureReason string
+
+	if isAborted {
+		failureReason = "aborted"
+
+		conditionManager := controller.NewVirtualMachineInstanceMigrationConditionManager()
+		if !conditionManager.HasCondition(migration, virtv1.VirtualMachineInstanceMigrationAbortRequested) {
+			condition := virtv1.VirtualMachineInstanceMigrationCondition{
+				Type:          virtv1.VirtualMachineInstanceMigrationAbortRequested,
+				Status:        k8sv1.ConditionTrue,
+				LastProbeTime: v1.Now(),
+			}
+			migration.Status.Conditions = append(migration.Status.Conditions, condition)
+		}
+
+		c.recorder.Eventf(migration, k8sv1.EventTypeNormal, SuccessfulAbortMigrationReason, "migration is successfully aborted")
+
+	} else {
+		failureReason = "failed"
+
+		c.recorder.Eventf(migration, k8sv1.EventTypeWarning, FailedMigrationReason, "source node reported migration failed")
+		handleFailedMigrationMetrics(migration, vmi, pod)
+	}
+
+	log.Log.Object(migration).Errorf("VMI %s/%s reported migration %s", vmi.Namespace, vmi.Name, failureReason)
+}
+
+func (c *MigrationController) markMigrationAbortInVmiStatus(migration *virtv1.VirtualMachineInstanceMigration, vmi *virtv1.VirtualMachineInstance) error {
 
 	vmiCopy := vmi.DeepCopy()
+	now := v1.NewTime(time.Now())
 
+	if vmiCopy.Status.MigrationState == nil {
+		vmiCopy.Status.MigrationState = &virtv1.VirtualMachineInstanceMigrationState{
+			MigrationUID:   migration.UID,
+			SourceNode:     vmi.Status.NodeName,
+			StartTimestamp: &now,
+		}
+	}
```

Comment:
you should never need to do this during the abort... If the migration was never handed off to virt-handler ```vmiCopy.Status.MigrationState == nil``` then there's no reason to modify the VMI at all.

Reply:
Thanks! You're right, these are leftover.